lib/frameworks/jasmine2.js

```diff
@@ -74,11 +74,29 @@ exports.run = function(runner, specs) {
   var reporter = new RunnerReporter(runner);
   jasmine.getEnv().addReporter(reporter);
 
-  // Filter specs to run based on jasmineNodeOpts.grep and jasmineNodeOpts.invert.
-  jasmine.getEnv().specFilter = function(spec) {
+  // Filter specs/suite to run based on jasmineNodeOpts.grep and jasmineNodeOpts.invert.
+  jasmine.getEnv().specFilter = function(spec, suite) {
+    var suiteMatched = isSuiteMatched(suite);
+
+    //If it matched at suite level and invert is true, all the specs will not be included.
+    if(suiteMatched && jasmineNodeOpts.invertGrep){
```

Comment:
if the filter is applied at suite level, it means that it will apply for all specs under the Suite.